crates/services/txpool_v2/src/extracted_outputs.rs

```diff
@@ -0,0 +1,119 @@
+//! Temporary module to extract outputs until we know that they have been settled in database or skipped.
+
+use std::collections::HashMap;
+
+use fuel_core_types::{
+    fuel_tx::{
+        input::coin::{
+            CoinPredicate,
+            CoinSigned,
+        },
+        Address,
+        AssetId,
+        ContractId,
+        Input,
+        Output,
+        TxId,
+        UtxoId,
+    },
+    services::txpool::ArcPoolTx,
+};
+
+pub struct ExtractedOutputs {
+    contract_created: HashMap<ContractId, TxId>,
+    coins_created: HashMap<TxId, HashMap<UtxoId, (Address, u64, AssetId)>>,
+}
+
+impl ExtractedOutputs {
+    pub fn new() -> Self {
+        Self {
+            contract_created: HashMap::new(),
+            coins_created: HashMap::new(),
+        }
+    }
+}
+
+impl Default for ExtractedOutputs {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl ExtractedOutputs {
+    pub fn new_extracted_transaction(&mut self, tx: &ArcPoolTx) {
+        let tx_id = tx.id();
+        for (idx, output) in tx.outputs().iter().enumerate() {
+            match output {
+                Output::ContractCreated { contract_id, .. } => {
+                    self.contract_created.insert(*contract_id, tx_id);
+                }
+                Output::Coin {
+                    to,
+                    amount,
+                    asset_id,
+                } => {
+                    self.coins_created.entry(tx_id).or_default().insert(
+                        UtxoId::new(
+                            tx_id,
+                            u16::try_from(idx)
+                                .expect("Outputs count is less than u16::MAX"),
+                        ),
+                        (*to, *amount, *asset_id),
+                    );
+                }
+                Output::Contract { .. }
+                | Output::Change { .. }
+                | Output::Variable { .. } => {
+                    continue;
+                }
+            }
+        }
+        for input in tx.inputs() {
+            match input {
+                Input::CoinSigned(CoinSigned { utxo_id, .. })
+                | Input::CoinPredicate(CoinPredicate { utxo_id, .. }) => {
+                    self.coins_created
+                        .entry(*utxo_id.tx_id())
+                        .or_default()
+                        .remove(utxo_id);
```

Comment:
`or_default` will create a trash in the memory, so if it already `Vacant`, you can just do nothing.

Reply:
Done